sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -287,194 +437,404 @@ public class EventProcessorClient : EventProcessorBase<PartitionContext>, IAsync
-            // behavior.  We could take a connection factory here instead of a single connection.
+            // TODO: we probably can remove this constructor because the processor does not have a single connection anymore.
+            // In fact, returning the same connection from the factory might result in undefined behavior.  We could take a
+            // connection factory here instead of a single connection.
 
             Argument.AssertNotNullOrEmpty(consumerGroup, nameof(consumerGroup));
             Argument.AssertNotNull(partitionManager, nameof(partitionManager));
             Argument.AssertNotNull(connection, nameof(connection));
 
             clientOptions = clientOptions?.Clone() ?? new EventProcessorClientOptions();
 
-            OwnsConnection = false;
             ConnectionFactory = () => connection;
             FullyQualifiedNamespace = connection.FullyQualifiedNamespace;
             EventHubName = connection.EventHubName;
             ConsumerGroup = consumerGroup;
             Manager = partitionManager;
-            Options = clientOptions;
+            ClientOptions = clientOptions;
             RetryPolicy = clientOptions.RetryOptions.ToRetryPolicy();
-            Identifier = Guid.NewGuid().ToString();
-        }
-
-        /// <summary>
-        ///   Initializes a new instance of the <see cref="EventProcessorClient"/> class.
-        /// </summary>
-        ///
-        protected EventProcessorClient()
-        {
-            OwnsConnection = false;
+            Identifier = string.IsNullOrEmpty(clientOptions.Identifier) ? Guid.NewGuid().ToString() : clientOptions.Identifier;
         }
 
         /// <summary>
-        ///   The function to be called just before event processing starts for a given partition.
+        ///   Starts the event processor.  In case it's already running, nothing happens.
         /// </summary>
         ///
-        /// <param name="context">The context in which the associated partition will be processed.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
         ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
-        protected override async ValueTask InitializeProcessingForPartitionAsync(PartitionContext context)
+        /// <exception cref="EventHubsClientClosedException">Occurs when this <see cref="EventProcessorClient" /> instance is already closed.</exception>
+        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="ProcessEventAsync" /> or <see cref="ProcessErrorAsync" /> set.</exception>
+        ///
+        public virtual async Task StartProcessingAsync(CancellationToken cancellationToken = default)
         {
-            try
+            Argument.AssertNotClosed(IsClosed, nameof(EventProcessorClient));
```

Comment:
Since we're removing `IsClosed` should we use `IsRunning` here or is this no longer needed?

Reply:
No longer needed. We're supposed to simply ignore the call if the processor is running.